src/utils/Git.ts

```diff
@@ -121,12 +122,20 @@ export async function branches({
     remotes,
     tags,
 }: BranchesOptions): Promise<Branch[]> {
+    const currentBranch = await executeCommand(
+        "git",
+        ['"symbolic-ref"', '"--short"', '"-q"', '"HEAD"'],
+        directory)
+        .then(branchName => branchName.trim());
     let lines = await linedOutputOf(
         "git",
         ["for-each-ref", tags ? "refs/tags" : "", "refs/heads", remotes ? "refs/remotes" : "", "--format='%(HEAD)%(refname:short)'"],
         directory,
     );
-    return lines.map(line => new Branch(line.slice(1), line[0] === "*"));
+    return lines.map(line => {
```

Comment:
Can I be sure that both the "currentBranch" and "lines" asynchronous calls always finish before "lines.map" code runs?